patroni/postgresql/slots.py

```diff
@@ -42,9 +46,17 @@ def compare_slots(s1: Dict[str, Any], s2: Dict[str, Any], dbid: str = 'database'
 
 
 class SlotsAdvanceThread(Thread):
+    """Daemon process :class:``Thread`` object for advancing logical replication slots on replicas.
+
+    This ensures that slot advancing queries sent to postgres do not block the main loop and cause leader lock expiry.
```

Comment:
It runs only in replicas and therefore consequences are different:

0. Replica will not be able to participate in leader race
1. Member key for this node could expire
2. As a result the leader will start making attempts to remove the replication slot.
3. While physical replication is active the physical slot could not be removed (good)
4. but in case if replica for some reason disconnected the physical slot will be dropped (txid_current on the physical slot will stop preventing autovacuum from cleaning up dead tuples in pg_catalog).

```suggestion
    This ensures that slot advancing queries sent to postgres do not block the main loop.
```